lib/managers/collaboration-whitelists.js

```diff
@@ -0,0 +1,225 @@
+ *                                                      collaboration actions taken within and external to an enterprise
+ * @param {Function} callback - Passed the collaboration whitelist information if it was created successfully
+ * @returns {Promise<Object>} A promise resolve to the collaboration whitelist object
+ */
+CollaborationWhitelists.prototype.create = function(domain, direction, callback) {
+
+	var params = {
+		body: {
+			domain: domain,
+			direction: direction
+		}
+	};
+
+	var apiPath = urlPath(BASE_PATH);
+	return this.client.wrapWithDefaultHandler(this.client.post)(apiPath, params, callback);
+};
+
+/**
+ * Requests a collaboration whitelist object with a given ID.
+ *
+ * API Endpoint: '/collaboration_whitelist_entries/:collaborationWhitelistID'
+ * Method: GET
+ *
+ * @param {string} collaborationWhitelistID - Box ID of the collaboration whitelist being requested
+ * @param {Object} [qs] - Additional options can be passed with the request via querystring. Can be left null in most cases.
+ * @param {Function} [callback] - Passed the collaboration whitelist information if it was acquired successfully
+ * @returns {Promise<Object>} A promise resolving to the collaboration whitelist object
+ */
+CollaborationWhitelists.prototype.get = function(collaborationWhitelistID, qs, callback) {
+
+	var params = { qs };
+
+	var apiPath = urlPath(BASE_PATH, collaborationWhitelistID);
+	return this.client.wrapWithDefaultHandler(this.client.get)(apiPath, params, callback);
+};
+
+/**
+ * Requests all collaboration whitelist objects within an enterprise.
+ *
+ * API Endpoint: '/collaboration_whitelist_entries'
+ * Method: GET
+ *
+ * @param {Object} [options] - Additional options. Can be left null in most cases.
+ * @param {int} [options.limit] - The number of collaboration whitelists to retrieve
+ * @param {string} [options.marker] - Paging marker, retrieve records starting at this position in the list. Left blank to start at the beginning.
+ * @param {Function} [callback] - Passed a list of collaboration whitelists if successful, error otherwise
+ * @returns {Promise<Object>} A promise resolving to the collection of collaboration whitelists
+ */
+CollaborationWhitelists.prototype.getAll = function(options, callback) {
+
+	var params = {
+		qs: options
+	};
+
+	var apiPath = urlPath(BASE_PATH);
+	return this.client.wrapWithDefaultHandler(this.client.get)(apiPath, params, callback);
+};
+
+/**
+ * Delete a given collaboration whitelist.
+ *
+ * API Endpoint: '/collaboration_whitelist_entries/:collaborationWhitelistID'
+ * Method: DELETE
+ *
+ * @param {string} collaborationWhitelistID - Box ID of the collaboration whitelist being requested
+ * @param {Function} [callback] - Empty response body passed if successful.
+ * @returns {Promise<void>} A promise resolving to nothing
+ */
+CollaborationWhitelists.prototype.delete = function(collaborationWhitelistID, callback) {
```

Comment:
If we go with the suggestion above, maybe this should be `collaborationWhitelist.removeDomain` for parity